nixos/modules/services/web-servers/hydron.nix

```diff
@@ -80,69 +82,78 @@ in with lib; {
   };
 
   config = mkIf cfg.enable {
-    services.hydron.passwordFile = mkDefault (pkgs.writeText "hydron-password-file" cfg.password);
-    services.hydron.postgresArgsFile = mkDefault (pkgs.writeText "hydron-postgres-args" cfg.postgresArgs);
-    services.hydron.postgresArgs = mkDefault ''
-      {
-        "driver": "postgres",
-        "connection": "user=hydron password=${cfg.password} host=/run/postgresql dbname=hydron sslmode=disable"
-      }
-    '';
-
-    services.postgresql = {
-      enable = true;
-      ensureDatabases = [ "hydron" ];
-      ensureUsers = [
-        { name = "hydron";
-          ensurePermissions = { "DATABASE hydron" = "ALL PRIVILEGES"; };
-        }
-      ];
-    };
-
-    systemd.tmpfiles.rules = [
-      "d '${cfg.dataDir}' 0750 hydron hydron - -"
-      "d '${cfg.dataDir}/.hydron' - hydron hydron - -"
-      "d '${cfg.dataDir}/images' - hydron hydron - -"
-      "Z '${cfg.dataDir}' - hydron hydron - -"
-
-      "L+ '${cfg.dataDir}/.hydron/db_conf.json' - - - - ${cfg.postgresArgsFile}"
-    ];
-
-    systemd.services.hydron = {
-      description = "hydron";
-      after = [ "network.target" "postgresql.service" ];
-      wantedBy = [ "multi-user.target" ];
-
-      serviceConfig = {
-        User = "hydron";
-        Group = "hydron";
-        ExecStart = "${pkgs.hydron}/bin/hydron serve"
-        + optionalString (cfg.listenAddress != null) " -a ${cfg.listenAddress}";
+    services = {
+      hydron = {
+        passwordFile = mkDefault (pkgs.writeText "hydron-password-file" cfg.password);
+        postgresArgsFile = mkDefault (pkgs.writeText "hydron-postgres-args" cfg.postgresArgs);
+
+        postgresArgs = mkDefault ''
+          {
+            "driver": "postgres",
+            "connection": "user=hydron password=${cfg.password} host=/run/postgresql dbname=hydron sslmode=disable"
+          }
+        '';
       };
-    };
 
-    systemd.services.hydron-fetch = {
-      description = "Import paths into hydron and possibly fetch tags";
+      postgresql = {
+        enable = true;
+        package = pkgs.postgresql_11;
```

Comment:
I don't think this is a good idea. This essentially changes the postgres version for all NixOS services

Reply:
That's a good point for this package, but there's an issue with that when it comes to the meguca package, which can't use the default of... I think it was 9.6?

Reply:
Maybe have an assertion for the version being above X